File: code-samples/cyberpurple.dart
class CyberHacker {
  String location;
  int birthDate, deathDate;
  List<String> weaknesses;

  CyberHacker({this.location, this.birthDate, this.deathDate, this.weaknesses});

  int get age => this.calcAge();

  int calcAge() => this.deathDate - this.birthDate;
}

void main() {
  // ...there was a guy named Neo
  final CyberSavvy = CyberHacker(
      location: 'New Tokyo',
      birthDate: 2040,
      deathDate: 2080,
      weaknesses: ['Sunlight', 'Neural inhibitors']);
}

File: code-samples/cyberpurple.rb
class CyberDude
  attr_reader :location, :birthDate, :deathDate, :weaknesses

  def initialize(opts)
    @location = opts[:location]
    @birthDate = opts[:birthDate]
    @deathDate = opts[:deathDate]
    @weaknesses = opts[:weaknesses]
  end

  def age
    calcAge
  end

  private

  def calcAge
    @deathDate - @birthDate
  end
end

cyberDude = CyberDude.new(
  location: 'Neo-Tokyo',
  birthDate: 2000,
  deathDate: 2100,
  weaknesses: %w[Sunlight Garlic],
)

puts cyberDude.age
